src/plugins/pcr_primer_design_for_dna_assembly/src/tasks/PCRPrimerDesignForDNAAssemblyTask.cpp

```diff
@@ -544,30 +629,113 @@ QString PCRPrimerDesignForDNAAssemblyTask::getPairReport(U2Region forward,
             arg(forward.toString(), primerName).arg(sequence.length());
         SAFE_POINT(forward.startPos > 0 && forward.startPos < sequence.length() && forward.length > 0, error,
             report += tr("<h3>Error, see log.</h3>"))
 
-        report += QString("<div class=\"seq\"><u>%1</u><b>%2</b></div>").arg(QString(backboneSequence)).
-            arg(QString(sequence.mid(forward.startPos, forward.length)));
+        report += getForwardPrimerReportResult(backboneSequence, sequence.mid(forward.startPos, forward.length));
     }
     if (!reverse.isEmpty()) {
-        report += tr("<h2>%1 Reverse:</h2>").arg(primerName);
+        report += getPrimerReportTitle(primerName, false);
 
         QString error = tr("Invalid region %1 for %2 reverse sequence of length %3").
             arg(reverse.toString(), primerName).arg(sequence.length());
         SAFE_POINT(reverse.startPos > 0 && reverse.startPos < sequence.length() && reverse.length > 0, error,
             report += tr("<h3>Error, see log.</h3>"))
 
-        report += QString("<div class=\"seq\"><b>%1</b><u>%2</u></div>").
-            arg(QString(sequence.mid(reverse.startPos, reverse.length))).arg(QString(backboneSequence));
+        report += getReversePrimerReportResult(backboneSequence, sequence.mid(reverse.startPos, reverse.length));
     }
     return report;
 }
 
+QString PCRPrimerDesignForDNAAssemblyTask::getPairReportForUserPrimers() const {
+    QString report_;
+    if (!settings.forwardUserPrimer.isEmpty() && !settings.reverseUserPrimer.isEmpty()) {
+        report_ += getPrimerReportTitle("C", true);
+        report_ += getForwardPrimerReportResult(backboneSequence, settings.forwardUserPrimer);
+        report_ += getPrimerReportTitle("C", false);
+        report_ += getReversePrimerReportResult(backboneSequence, settings.reverseUserPrimer);
+    }
+    return report_;
+}
+
+QString PCRPrimerDesignForDNAAssemblyTask::getUserPrimersUnwantedConnectionsReport() const {
+    using Seq = SeqToSearchInThem;
+    const QList<Seq> allTypes = { Seq::ForwardUser, Seq::ReverseUser, Seq::Sequence, Seq::RevComplSeq, Seq::OtherSeq };
+    const QMap<Seq, QString> toString = { {Seq::ForwardUser, tr("User forward primer")},
+                                          {Seq::ReverseUser, tr("User reverse primer")},
+                                          {Seq::Sequence, tr("Sequence")},
+                                          {Seq::RevComplSeq, tr("Reverse complement sequence")},
+                                          {Seq::OtherSeq, tr("Other sequences")} };
+
+    // Summary table for unwanted connections. Rows -- user primers, columns -- sequences with which unwanted
+    // connections have been formed.
+    QString report_ = tr("<br><h3>Unwanted connections user primers</h3>");
+    QString table = "<table border=\"1\" cellpadding=\"4\">"
+                      "<tr>"
+                        "<td></td>"
+                        "<th>User forward</th>"
```

Comment:
The same as [here](https://github.com/ugeneunipro/ugene/pull/525/files#r690046092)